mesonbuild/msetup.py

```diff
@@ -169,16 +169,15 @@ def validate_dirs(self, dir1: str, dir2: str, reconfigure: bool, wipe: bool) ->
         has_partial_build = os.path.isdir(priv_dir)
         if has_valid_build:
             if not reconfigure and not wipe:
-                print('Directory already configured.\n'
-                      '\nJust run your build command (e.g. ninja) and Meson will regenerate as necessary.\n'
+                raise MesonException('Directory already configured.\n\n'
+                      'Just run your build command (e.g. ninja) and Meson will regenerate as necessary.\n'
```

Comment:
This eventuality MUST NOT be considered user error.

Reply:
Why not? That makes no sense.

Reply:
Hmm, looks like 5years ago that was a problem already: https://github.com/mesonbuild/meson/pull/4249. That still feels completely wrong IMHO.

Reply:
Ok, kept that as success for now because it's really unrelated to this PR anyway. But IMHO we should reconsider this in the future.